legion/legion/k8s/enclave.py

```diff
@@ -0,0 +1,410 @@
+
+    @property
+    def graphite_service(self):
+        """
+        Return Graphite service object.
+
+        :return: :py:class:`legion.k8s.Service` -- Graphite service object or None
+        """
+        self._load_services()
+        return self._graphite_service
+
+    @property
+    def models(self):
+        """
+        Return a dict of Models services.
+
+        :return: dict[:py:class:`legion.k8s.ModelIdVersion`, :py:class:`legion.k8s.ModelService`] -- deployed models
+        """
+        models = {}
+        for service in legion.k8s.utils.find_all_services(namespace=self.name):
+            if legion.k8s.services.ModelService.is_model_service(service):
+                model_service = legion.k8s.services.ModelService(service)
+                models[model_service.id_and_version] = model_service
+        return models
+
+    def get_models(self, model_id, model_version=None):
+        """
+        Get models that fit match criterions (model id and model version, model id may be *, model version may be *)
+
+        :param model_id: model id or * for all models
+        :type model_id: str
+        :param model_version: (Optional) model version or * for all
+        :type model_version: str
+        :return: list[:py:class:`legion.k8s.ModelService`] -- founded model services
+        """
+        return [
+            model_service
+            for id_and_version, model_service in self.models.items()
+            if model_id in (id_and_version.id, '*') and model_version in (id_and_version.version, None, '*')
+        ]
+
+    def get_models_strict(self, model_id, model_version=None):
+        """
+        Get models that fit match criterions (model id and model version, model id may be *, model version may be *)
+        If more that one model would be found with unstrict criterion - exception would be raised
+        If no one model would be found - exception would be raised
+
+        param model_id: model id or * for all models
+        :type model_id: str
+        :param model_version: (Optional) model version
+        :type model_version: str
+        :return: list[:py:class:`legion.k8s.ModelService`] -- founded model services
+        """
+        model_services = self.get_models(model_id, model_version)
+
+        if len(model_services) > 1 and not model_version:
+            raise Exception('Please specify version of model')
+
+        if not model_services:
+            raise Exception('No one model can be found')
+
+        return model_services
+
+    def deploy_model(self, image, count=1):
+        """
+        Deploy new model
+
+        :param image: docker image with model
+        :type image: str
+        :param count: count of pods
+        :type count: int
+        :return: :py:class:`legion.k8s.services.ModelService` -- model service
+        """
+        client = legion.k8s.utils.build_client()
+
+        introspection_image = image
+
+        if os.getenv('EXTERNAL_TEST') and '-local' in introspection_image:
```

Comment:
This looks like a hack for me.

Reply:
Yeah, we have that problem, because we have restrictions for connections

Reply:
I will try to configure firewall